pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -119,7 +120,46 @@ func shutdownAppsGlobal(ctx *zedagentContext) {
 	shutdownApps(ctx.getconfigCtx)
 }
 
-var baseosPrevConfigHash []byte
+var baseOSPrevConfigHash []byte
+
+func parseBaseOS(getconfigCtx *getconfigContext,
+	config *zconfig.EdgeDevConfig) {
+
+	baseOS := config.GetBaseos()
+	if baseOS == nil {
+		//unpublish if we received something before (have hash)
+		if len(baseOSPrevConfigHash) > 0 {
+			log.Function("parseBaseOS: nil config received: clear previous published one")
+			baseOSPrevConfigHash = []byte{}
+			getconfigCtx.pubBaseOs.Unpublish("global")
+		}
+		return
+	}
+	h := sha256.New()
+	computeConfigElementSha(h, baseOS)
+	configHash := h.Sum(nil)
+	same := bytes.Equal(configHash, baseOSPrevConfigHash)
+	if same {
+		return
+	}
+	log.Functionf("parseBaseOS: Applying updated config "+
+		"prevSha: % x, "+
+		"NewSha : % x, "+
+		"baseOS: %v",
+		baseOSPrevConfigHash, configHash, baseOS)
+	baseOSPrevConfigHash = configHash
+	var configRetryUpdateCounter uint32
+	if baseOS.GetRetryUpdate() != nil {
+		configRetryUpdateCounter = baseOS.GetRetryUpdate().GetCounter()
+	}
```

Comment:
Why do you need this conditional?
The counter is just baseOS.ConfigRetryUpdateCounter, right?

Reply:
It is the pointer (*DeviceOpsCmd), we must check it before accessing of nested structures

Reply:
Got it